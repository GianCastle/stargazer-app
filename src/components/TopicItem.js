import React from 'react'

export default function TopicItem({ topic, setSelectedcriteria }) {
  return (
    <li className="list-group-item" data-testid={`topic-item-${topic?.node?.resourcePath}`}>
      <div className="d-flex justify-content-between">
        <h5><strong>{topic?.node?.resourcePath}  <i className="fa fa-star" aria-hidden="true" />{Number(topic?.node?.stargazers?.totalCount)?.toLocaleString()}
        </strong></h5>
      </div>
      <div>
        {Boolean(!topic?.node?.repositoryTopics?.nodes?.length) && '❌ No topics found '}
        {Boolean(topic?.node?.repositoryTopics?.nodes?.length) && '💻 Repo topics -> '}

        {topic?.node?.repositoryTopics?.nodes?.map((node, j) => (
          <button key={j}
            onClick={() => setSelectedcriteria(node?.topic?.name)}
            type="button" className="btn  btn-outline-primary btn-sm mx-1 my-1">{node?.topic?.name} <span className="badge badge-light badge-pill"><i className="fa fa-star m1-2" aria-hidden="true" />{node?.topic?.stargazerCount}</span></button>
        )) ?? 'no topics'}
      </div>
    </li>
  )
}