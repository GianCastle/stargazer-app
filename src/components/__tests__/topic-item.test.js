import { render } from "@testing-library/react";
import TopicItem from "../TopicItem";


describe('header component', () => {
  it('should render sucessfully a topic', () => {
    const mockedTopic = {
      node: {
        resourcePath: "test-12",
        stargazers: {
          totalCount: 1
        },
        repositoryTopics: {
          totalCount: 1,
          nodes: [
            {
              topic: {
                name: "test",
                stargazerCount: 1,
                relatedTopics: [
                  {
                    name: "test"
                  }
                ]
              }
            }
          ]

        }
      }
    }
    const { wrapper, getByTestId } = render(<TopicItem topic={mockedTopic} />);
    const header = getByTestId('topic-item-test-12');

    expect(header).toHaveTextContent('test-12 1üíª Repo topics -> test 1');
    expect(wrapper).toMatchSnapshot();
  });

  it('should contains no-topic labels if no labels for a repo is found', () => {
    const mockedTopic = {
      node: {
        resourcePath: "test-12",
        stargazers: {
          totalCount: 1
        },
        repositoryTopics: {
          totalCount: 0,
          nodes: []
        }
      }
    }
    const { wrapper, getByTestId } = render(<TopicItem topic={mockedTopic} />);
    const header = getByTestId('topic-item-test-12');

    expect(header).toHaveTextContent('test-12 1‚ùå No topics found');
    expect(wrapper).toMatchSnapshot();
  });
})