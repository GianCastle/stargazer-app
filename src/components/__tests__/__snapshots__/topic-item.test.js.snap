// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component should contains no-topic labels if no labels for a repo is found 1`] = `undefined`;

exports[`header component should render sucessfully a topic 1`] = `undefined`;
